### PR TITLE
feat(ImageBlock): respect downloadability in asset viewer

### PR DESCRIPTION
**Pull request by fulopdaniel**

### CODE REVIEW

File: packages/image-block/src/ImageBlock.tsx [hunk 1]
```diff
@@ -56,6 +57,12 @@ export const ImageBlock = withAttachmentsProvider(({ appBridge }: BlockProps) =>
     const { blockAssets, deleteAssetIdsFromKey, updateAssetIdsFromKey } = useBlockAssets(appBridge);
     const image = blockAssets?.[IMAGE_ID]?.[0];
     const attachmentCount = blockAssets[ATTACHMENTS_ASSET_ID]?.length || 0;
+    const isImageDownloadable = isDownloadable(
```
Comment (ragi96) [inline packages/image-block/src/ImageBlock.tsx:60]: add here also the check if its a protected image, 

If its protected always false, otherewise determine based on the `isDownloadable`

We will move that logic in the long run into the `guideline-blocks-settings` but lets wait, so its not becoming blocking

Comment (fulopdaniel) [reply packages/image-block/src/ImageBlock.tsx]: Yep, updated!

File: packages/image-block/src/ImageBlock.tsx [hunk 2]
```diff
@@ -56,6 +57,10 @@ export const ImageBlock = withAttachmentsProvider(({ appBridge }: BlockProps) =>
     const { blockAssets, deleteAssetIdsFromKey, updateAssetIdsFromKey } = useBlockAssets(appBridge);
     const image = blockAssets?.[IMAGE_ID]?.[0];
     const attachmentCount = blockAssets[ATTACHMENTS_ASSET_ID]?.length || 0;
+    const isImageDownloadable =
+        image?.isDownloadProtected === false &&
+        isDownloadable(blockSettings.security, blockSettings.downloadable, assetDownloadEnabled);
```
Comment (ragi96) [inline packages/image-block/src/ImageBlock.tsx:62]: I think it would make sense to move that into the helpers and add some unit tests for it

Comment (ragi96) [reply packages/image-block/src/ImageBlock.tsx:62]: And as i just saw it the helper that is already there, should move into the `shared`

Comment (fulopdaniel) [reply packages/image-block/src/ImageBlock.tsx]: Updated, and added some tests to test the new logic, as the isDownloadable is already tested in `guideline-blocks-settings`

**Review by ragi96 — APPROVED** [review]
LGTM

**Pull request merged**